### PR TITLE
refactor: refactored ProfilePage.tsx to be functional component

### PR DESCRIPTION
**Pull request by Alphajax**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [x] I followed naming convention rules

---

#### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [x] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 💡 Background and solution

Refactor [ProfilePage](https://github.com/rolling-scopes/rsschool-app/blob/master/client/src/pages/profile/index.tsx) to be functional component
Use `useState` for state management
Use `useEffect` instead of lifecycle methods
Use `SessionContext` instead of `withSession` HOC
Use `SessionContext` instead of `withSession` HOC at Job page
Removed `withSession` function which is not used any more

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Database migration is added or not needed
- [x] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

**Comment by valerydluski** [comment]
Could you also check if we still need this code after your changes?
https://github.com/rolling-scopes/rsschool-app/blob/master/client/src/components/withSession.tsx#L35

File: client/src/pages/profile/index.tsx [hunk 1]
```diff
@@ -22,53 +21,38 @@ import PreScreeningIviewCard from 'components/Profile/PreScreeningIviewCard';
...
+  const router = useRouter();
+  const session = useContext(SessionContext);
+  const [profile, setProfile] = useState<ProfileInfo | null>(null);
+  const [isProfileOwner, setIsProfileOwner] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isSaving, setIsSaving] = useState(false);
+  const [connections, setConnections] = useState<Connections>({});
 
-  private hadStudentCoreJSInterview = (stats: StudentStats[]) =>
+  const hadStudentCoreJSInterview = (stats: StudentStats[]) =>
```
Comment (I-vasilich-I) [inline client/src/pages/profile/index.tsx:52]: I'd move helper functions(which do not rely on state/props) outside of the component.

File: client/src/pages/profile/index.tsx [hunk 2]
```diff
@@ -107,210 +91,180 @@ export class ProfilePage extends React.Component<Props, State> {
...
+  const isAdmin = session.isAdmin;
 
-    const cards = [
-      profile?.generalInfo && (
-        <MainCard data={mainInfo} isEditingModeEnabled={isProfileOwner} updateProfile={this.updateProfile} />
-      ),
-      <AboutCard
-        key="about-card"
-        data={aboutMyself}
+  const cards: JSX.Element[] = [
```
Comment (I-vasilich-I) [inline client/src/pages/profile/index.tsx:158]: you not only setting type here, but also casting it again at the end) 
though ts would know the type even if you don't set it at all.

File: client/src/pages/profile/index.tsx [hunk 3]
```diff
@@ -107,210 +91,180 @@ export class ProfilePage extends React.Component<Props, State> {
...
+              `}</style>
+            </div>
+          ) : (
+            <>
+              <Result status={'403' as any} title="No access or user does not exist" />
+            </>
+          )}
+        </Spin>
+      </LoadingScreen>
+    </>
```
Comment (I-vasilich-I) [inline client/src/pages/profile/index.tsx:256]: you don't need that fragment.

File: client/src/pages/profile/index.tsx [hunk 4]
```diff
@@ -87,16 +71,16 @@ export class ProfilePage extends React.Component<Props, State> {
...
+  };
 
+  const fetchData = async () => {
     try {
       const githubId = router.query ? (router.query.githubId as string) : undefined;
       const [profile, connections, { data }] = await Promise.all([
-        this.userService.getProfileInfo(githubId?.toLowerCase()),
-        this.notificationsService.getUserConnections().catch(() => []),
+        userService.getProfileInfo(githubId?.toLowerCase()),
+        notificationsService.getUserConnections().catch(() => []),
```
Comment (I-vasilich-I) [inline client/src/pages/profile/index.tsx:83]: array seems not the right thing to return here, connections is an object, well in js everything is an object), but maybe it should be actual object `({})` or `({} as object)`

File: client/src/pages/profile/index.tsx [hunk 5]
```diff
@@ -107,210 +91,180 @@ export class ProfilePage extends React.Component<Props, State> {
...
+              <style jsx global>{`
+                .masonry-column {
+                  padding-left: 16px;
+                  background-clip: padding-box;
+                }
+              `}</style>
+            </div>
+          ) : (
+            <>
+              <Result status={'403' as any} title="No access or user does not exist" />
```
Comment (I-vasilich-I) [inline client/src/pages/profile/index.tsx:251]: you can get rid of `any` here, just use `status="403"`

**Review by apalchys — APPROVED** [review]
Please fix comments by @I-vasilich-I

**Review by I-vasilich-I — APPROVED** [review]

**Review by valerydluski — APPROVED** [review]

**Pull request merged**